### PR TITLE
architecture: spmc: document raw binary format SPs

### PR DESCRIPTION
**Pull request by imre-kis-arm**
A new raw binary format was introduced to have a common secure partition format across different SPMC implementations. Add documentation about the format and its load process.

The PR that introduces the feature to optee_os is available here: https://github.com/OP-TEE/optee_os/pull/5794

### CODE REVIEW

**Comment by jbech-linaro** [comment]
@imre-kis-arm , sorry for delaying this. I'll have a look in the coming day(s).

**Review by jenswi-linaro — COMMENTED** [review]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: architecture/spmc.rst [hunk 1]
```diff
@@ -53,11 +53,37 @@ The
...
+
+SPMC agnostic flat binary format
+--------------------------------
+
+This simple binary format aims for maximum portability between SPMC
+implementations by removing the dependency on an ELF loader and implementation
+specific metadata in the SP image. The SPMC can simply copy the binary into the
+memory and start running it. The relocations, the stack setup and any further
+initialization steps should be handled by the startup code of the secure
+partition. The access rights for different section of the binary can be
```
Comment (etienne-lms) [inline architecture/spmc.rst:78]: s/section/sections/

File: architecture/spmc.rst [hunk 2]
```diff
@@ -89,8 +115,10 @@ This is done by adding ``sp_init_all()`` to the ``boot_final`` initcall level.
 				World.
 
 
-Each SP is loaded into the system using ``ldelf`` and started. This is based
-around the same process as loading the early TAs.
+Each ELF format SP is loaded into the system using ``ldelf`` and started. This
+is based around the same process as loading the early TAs.
+For each binary format SPs a simpler method is used to copy the binary into a
```
Comment (etienne-lms) [inline architecture/spmc.rst:120]: s/SPs/SP/

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

File: architecture/spmc.rst (hunk 1 above)
Comment (imre-kis-arm) [reply architecture/spmc.rst:78]: Done.

File: architecture/spmc.rst (hunk 2 above)
Comment (imre-kis-arm) [reply architecture/spmc.rst:120]: Done.

**Comment by jbech-linaro** [comment]
Looking good, thanks! Merging!

**Comment by imre-kis-arm** [comment]
Thanks!

**Pull request merged**